src/webapp/components/table/statistic-table/StatisticTable.tsx

```diff
@@ -0,0 +1,192 @@
+import React, { useCallback, useState } from "react";
+import styled from "styled-components";
+import {
+    Table,
+    TableBody,
+    TableCell,
+    TableHead,
+    TableRow,
+    TableContainer,
+} from "@material-ui/core";
+import { SearchInput } from "../../search-input/SearchInput";
+import i18n from "../../../../utils/i18n";
+import { MedianRow } from "./MedianRow";
+import { PercentTargetMetRow } from "./PercentTargetMetRow";
+import { MultipleSelector } from "../../selector/MultipleSelector";
+import { useTableFilters } from "../../../hooks/useTableFilters";
+import { useTableCell } from "../../../hooks/useTableCell";
+import { ColoredCell } from "./ColoredCell";
+import _ from "../../../../domain/entities/generic/Collection";
+
+export type TableColumn = {
+    value: string;
+    label: string;
+    dark?: boolean;
+};
+
+export type FilterType = {
+    value: TableColumn["value"];
+    label: TableColumn["label"];
+    type: "multiselector" | "datepicker";
+};
+
+export type FiltersValuesType = {
+    [key: TableColumn["value"]]: string[];
+};
+
+export type StatisticTableProps = {
+    columns: TableColumn[];
+    columnRules: {
+        [key: TableColumn["value"]]: number;
+    };
+    editRiskAssessmentColumns: TableColumn["value"][];
+    rows: {
+        [key: TableColumn["value"]]: string;
+    }[];
+    filters: FilterType[];
+};
+
+export const StatisticTable: React.FC<StatisticTableProps> = React.memo(
+    ({ rows, columns, columnRules, editRiskAssessmentColumns, filters: filtersConfig }) => {
+        const [searchTerm, setSearchTerm] = useState<string>("");
+
+        const [filters, setFilters] = useState<FiltersValuesType>(
+            filtersConfig.reduce((acc: FiltersValuesType, filter) => {
+                acc[filter.value] = [];
+                return acc;
+            }, {})
+        );
+
+        const calculateColumns = [...editRiskAssessmentColumns, ...Object.keys(columnRules)];
+
+        const filteredRows = useTableFilters(rows, searchTerm, filters);
```

Comment:
I would add to `useTableFilters` everything related to filters (search state, filters state and filterOptions function) to encapsulate the filter logic in the custom hook, and the hook will return everything that the component needs to work

Reply:
Sure StatisticTable logic will be lighter 👍